libtenzir/src/concept/parseable/tenzir/expression.cpp

```diff
@@ -219,10 +220,10 @@ auto make_predicate_parser() {
     | "!in"_p ->* [] { return relational_operator::not_in; }
     | "ni"_p  ->* [] { return relational_operator::ni; }
     | "!ni"_p ->* [] { return relational_operator::not_ni; }
-    | "[+"_p  ->* [] { return relational_operator::in; }
-    | "[-"_p  ->* [] { return relational_operator::not_in; }
-    | "+]"_p  ->* [] { return relational_operator::ni; }
-    | "-]"_p  ->* [] { return relational_operator::not_ni; }
+    | "not"_p >> required_ws_or_comment >> "in"_p 
+      ->* [] { return relational_operator::not_in; }
+    | "not"_p >> required_ws_or_comment >> "ni"_p 
+      ->* [] { return relational_operator::not_ni; }
```

Comment:
```suggestion
    | "not"_p >> required_ws_or_comment >> "in"_p 
      ->* [] { return relational_operator::not_in; }
```

Should be `ni ton` instead of `not ni`... Only slightly joking. But since it won't be valid in the new language, there's no need to add this, right?

Reply:
It really should be contains rather than ni, but yeah, I do agree.